app/actions/ModActionPanel/QuickAction.tsx

```diff
@@ -659,10 +658,11 @@ function Form(
                       name="labels"
                       formId={FORM_ID}
                       defaultLabels={currentLabels.filter((label) => {
-                        const isExternalLabel = allLabels.some((l) => {
-                          return l.val === label && l.src !== config.did
+                        // If there's a label where the source is the current labeler, it's editable
+                        const isEditableLabel = allLabels.some((l) => {
+                          return l.val === label && l.src === config.did
                         })
-                        return !isSelfLabel(label) && !isExternalLabel
+                        return !isSelfLabel(label) && isEditableLabel
```

Comment:
Is there any chance of `isEditableLabel` being true for a self-label?  Perhaps this could be simplified down to just `isEditableLabel`.

Reply:
iirc, the selfLabel check prevents labels that look like `porn(self)` to show up in the selector so we would still want to filter those out.